src/index.ts

```diff
@@ -11,16 +11,28 @@ function valueIsUnchanged(field: HTMLInputElement | HTMLTextAreaElement): boolea
 
 type StorageFilter = (field: HTMLInputElement | HTMLTextAreaElement) => boolean
 
-type PersistOptions = {
+type PersistOptionsWithSelector = {
+  scope?: ParentNode
   selector?: string
+  fields?: never
+}
+
+type PersistOptionsWithFields = {
+  fields?: NodeList | Node[]
+  selector?: never
+}
+
+type PersistOptions = (PersistOptionsWithSelector | PersistOptionsWithFields) & {
   keyPrefix?: string
   storage?: Pick<Storage, 'getItem' | 'setItem'>
   storageFilter?: StorageFilter
 }
 
 // Write all ids and values of the selected fields on the page into sessionStorage.
 export function persistResumableFields(id: string, options?: PersistOptions): void {
+  const scope = options && 'scope' in options ? options.scope ?? document : document
```

Comment:
I think this could be simplified to 

```suggestion
  const scope = options?.scope ?? document
```

you may have been fighting against the types but my suggestion earlier of adding the `scope?: never` should help